File: buzu/src/test/java/br/net/buzu/metadata/BasicMetadataTest.java
package br.net.buzu.metadata;

import br.net.buzu.metadata.build.parse.BasicMetadataParser;
import br.net.buzu.pplspec.model.Metadata;
import br.net.buzu.pplspec.model.PplString;
import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

/**
 * @author Douglas Siviotti
 *
 */
public class BasicMetadataTest {

	@Test
	public void test() {
		PplString pplString = new PplString("(name:S10;age:I3;city:S10)");
		Metadata bm1 = new BasicMetadataParser().parse(pplString);
		Metadata bm2 = new BasicMetadataParser().parse(pplString);
		
		assertTrue(bm1.equals(bm2));
		assertTrue(bm2.equals(bm1));
		assertTrue(bm2.hashCode() == bm1.hashCode());
		assertTrue(bm2.toString().equals(bm1.toString()));
		assertTrue(bm2.toTree(0).equals(bm1.toTree(0)));
		assertTrue(bm1.equals(bm1));
		assertFalse(bm1.equals(null));
		Object o = "abc";
		assertFalse(bm1.equals(o));
	}

}

File: ppl-lib/src/test/kotlin/br/net/buzu/pplimpl/metadata/BasicMetadataTest.kt
package br.net.buzu.pplimpl.metadata

import br.net.buzu.pplspec.model.pplStringOf
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

/**
 * @author Douglas Siviotti
 */
class BasicMetadataTest {

    @Test
    fun test() {
        val pplString = pplStringOf("(name:S10;age:I3;city:S10)")
        val bm1 = parseMetadata(pplString)
        val bm2 = parseMetadata(pplString)

        assertTrue(bm1 == bm2)
        assertTrue(bm2 == bm1)
        assertTrue(bm2.hashCode() == bm1.hashCode())
        assertTrue(bm2.toString() == bm1.toString())
        assertTrue(bm1 == bm1)
        assertFalse(bm1 == null)
        val o = "abc"
        assertFalse(bm1.equals(o))
    }

}
